src/main/java/com/google/ehub/servlets/ProfileServlet.java

```diff
@@ -65,10 +66,14 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     } else {
       String userEmail = userService.getCurrentUser().getEmail();
       UserProfile userProfile = profileData.getUserProfile(userEmail);
-
+      JsonObject profileJson = new JsonObject();
+      
       if (userProfile == null) {
-        response.sendRedirect("/CreateProfilePage.html");
-      } else {
+        profileJson.addProperty("NeedsProfile", true);
```

Comment:
Lets create a constant for "NeedsProfile" similar to the other other properties.